keps/sig-auth/20190607-certificates-api.md

```diff
@@ -292,8 +293,33 @@ type CertificateSigningRequestStatus struct {
   // Conditions applied to the request, such as approval or denial.
   Conditions []CertificateSigningRequestCondition
 
-  // If request was approved, the controller will place the issued
-  // certificate here.
+  // Certificate is populated by the signer with the issued certificate in PEM format.
+  // In JSON and YAML output, this entire field is base64-encoded, so it consists of:
+  // base64(
+  // -----BEGIN CERTIFICATE-----
+  // MII...Pb7Yu/E=
+  // -----END CERTIFICATE-----
+  // optional intermediate certificate blocks
+  // -----BEGIN CERTIFICATE-----
+  // MII...MGKB
+  // -----END CERTIFICATE-----
+  // -----BEGIN CERTIFICATE-----
+  // MII...AY1M
+  // -----END CERTIFICATE-----
+  // )
+  //
+  // In v1beta1, this field is unvalidated.
+  //
+  // In v1, modified content in this field is validated:
+  // * Content must contain one or more PEM blocks
```

Comment:
Since we are revisiting this: Why not make this `Certificates: [][]byte` that contains the der encoded certificates?

Reply:
I assume you mean changing the structure in v1; how would you envision v1beta1 <-> v1 round-tripping working?

Reply:
> Since we are revisiting this: Why not make this `Certificates: [][]byte` that contains the der encoded certificates?

I generally have trouble dealing with those both on disk and in the golang stack at runtime. We use PEM for everything I can think of that we read.

Reply:
SGTM